cmake/OpenCVModule.cmake

```diff
@@ -858,10 +858,14 @@ macro(_ocv_create_module)
 
   if((NOT DEFINED OPENCV_MODULE_TYPE AND BUILD_SHARED_LIBS)
       OR (DEFINED OPENCV_MODULE_TYPE AND OPENCV_MODULE_TYPE STREQUAL SHARED))
-    set_target_properties(${the_module} PROPERTIES COMPILE_DEFINITIONS CVAPI_EXPORTS)
+    target_compile_definitions(${the_module} PRIVATE CVAPI_EXPORTS)
     set_target_properties(${the_module} PROPERTIES DEFINE_SYMBOL CVAPI_EXPORTS)
   endif()
 
+  if (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_BUILD_TYPE STREQUAL "Debug")
+    target_compile_definitions(${the_module} PUBLIC _GLIBCXX_DEBUG)
+  endif()
```

Comment:
Is STL debug mode ABI compatible with non-debug mode?
If no, this may be a problem: this flag is added for modules itself only.
What is about compilation of user applications via OpenCV distribution package?